spikeextractors/extractors/cedextractors/cedrecordingextractor.py

```diff
@@ -0,0 +1,140 @@
+from spikeextractors import RecordingExtractor
+from .utils import get_channel_info, get_channel_data
+from spikeextractors.extraction_tools import check_get_traces_args
+
+import numpy as np
+from pathlib import Path
+from typing import Union
+
+try:
+    from sonpy import lib as sp
+
+    HAVE_SONPY = True
+except ImportError:
+    HAVE_SONPY = False
+
+PathType = Union[str, Path, None]
+DtypeType = Union[str, np.dtype, None]
+
+
+class CEDRecordingExtractor(RecordingExtractor):
+    """
+    Extracts electrophysiology recordings from .smrx files.
+    The recording extractor always returns channel IDs starting from 0.
+    The recording data will always be returned in the shape of (num_channels,num_frames).
+
+    Parameters
+    ----------
+    file_path: str
+        Path to the .smrx file to be extracted
+    smrx_ch_inds: list of int
+        List with indexes of valid smrx channels. Does not match necessarily
+        with extractor id.
+    """
+
+    extractor_name = 'CEDRecordingExtractor'
+    installed = HAVE_SONPY  # check at class level if installed or not
+    is_writable = False
+    mode = 'file'
+    installation_mesg = "To use the CED extractor, install sonpy: \n\n pip install sonpy\n\n"  # error message when not installed
+
+    def __init__(self, file_path: PathType, smrx_ch_inds: list):
+        assert HAVE_SONPY, self.installation_mesg
+        file_path = Path(file_path)
+        assert file_path.is_file() and file_path.suffix == '.smrx', 'file_path must lead to a .smrx file!'
+
+        RecordingExtractor.__init__(self)
```

Comment:
```suggestion
        super().__init__(self)
```

Reply:
@luiztauffer not really that important but I was curious why you didn't use the `super` syntax here

Reply:
@bendichter this is usually what we used in spikeextractors. I agree we could switch to `super()` for clarity